app/src/main/java/be/brunoparmentier/openbikesharing/app/activities/MapActivity.java

```diff
@@ -96,6 +99,16 @@ protected void onCreate(Bundle savedInstanceState) {
         stationsDataSource = new StationsDataSource(this);
         ArrayList<Station> stations = stationsDataSource.getStations();
 
+        final Context context = getApplicationContext();
+        Configuration.getInstance().load(context, PreferenceManager.getDefaultSharedPreferences(context));
+        int versionCode = 0;
+        try {
+            versionCode = context.getPackageManager().getPackageInfo(getPackageName(), 0).versionCode;
+        } catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+        }
+        Configuration.getInstance().setUserAgentValue(context.getPackageName() + "/" + versionCode);
+
```

Comment:
I don't think we really need the versionCode in the UserAgent + I'd use `BuildConfig.APPLICATION_ID` which identifies the app, instead of `context.getPackageName()` (the package name can be different from the application ID).

So I'd replace the whole block with:
```
Configuration.getInstance().setUserAgentValue(BuildConfig.APPLICATION_ID)
```
(as suggested in https://github.com/osmdroid/osmdroid/wiki/Important-notes-on-using-osmdroid-in-your-app#set-the-http-user-agent-variable)

Reply:
Hi, yeah it's true that the osmdroid does not officially recommend to go as far as this. Actually, as stated in my commit message I relied on the osmdroid github thread that pointed the issue, and more precisely a comment of an OSM sysadmin called Firefishy.
My understanding of the whole bug is that, at some point of the time when there was a single user-agent for every osmdroid client, it has been banned by the OSM Foundation because of overloading problems in which it was involved. Now the new osmdroid requires the customization of the user-agent so OSM could ban only the ending client that overloads its servers. In this context using the APPLICATION_ID would be enough. That said, the recommendation of Firefishy was to use a different user-agent for each app version to prevent the whole package to be banned forever if a specific version behaves badly for some reason.